sklearn/linear_model/stochastic_gradient.py

```diff
@@ -714,9 +714,14 @@ def __init__(self, loss="hinge", penalty='l2', alpha=0.0001, l1_ratio=0.15,
             average=average)
 
     def _check_proba(self):
-        check_is_fitted(self, "t_")
-
-        if self.loss not in ("log", "modified_huber"):
+        try:
+            check_is_fitted(self, ["t_", "coef_", "intercept_"],
+                            all_or_any=all)
+            is_fitted = True
+        except _NotFittedError:
+            is_fitted = False            
```

Comment:
I am not sure it's the right fix.

is check_is_fitted really needed to know if predict_proba is supported?

can't you just remove the call to check_is_fitted ?